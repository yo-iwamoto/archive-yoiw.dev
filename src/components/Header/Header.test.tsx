import { Header } from '.';
import { render } from '@testing-library/react';

describe.skip('components/Header', () => {
  it('正しく表示されること', () => {
    const { getByRole, getAllByRole, asFragment } = render(<Header />);

    expect(getByRole('banner', { name: '' })).toBeDefined();
    expect(getByRole('link', { name: 'yoiw.dev' })).toBeDefined();
    expect(getByRole('button', { name: 'メニューを開く' })).toBeDefined();

    // tailwind で PC/SP 表示を制御しているため、以下は2つずつ存在する
    expect(getAllByRole('navigation')).toHaveLength(2);
    expect(getAllByRole('link', { name: 'Blog' })).toHaveLength(2);
    expect(getAllByRole('link', { name: 'Resume' })).toHaveLength(2);

    expect(asFragment()).toMatchSnapshot();
  });
});
