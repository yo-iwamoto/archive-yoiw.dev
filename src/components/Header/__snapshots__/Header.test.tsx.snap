// Vitest Snapshot v1

exports[`components/Header > 正しく表示されること 1`] = `
<DocumentFragment>
  <header
    class="sticky top-0 bg-slate-700 py-2 px-4 md:py-4 md:px-12"
  >
    <div
      class="mx-auto flex max-w-7xl items-center justify-between"
    >
      <a
        class="text-xl font-bold text-white md:text-3xl"
        href="/"
      >
        yoiw.dev
      </a>
      <div
        class="md:hidden"
      >
        <button
          aria-label="メニューを開く"
          type="button"
        >
          <div
            aria-hidden="true"
            class="relative flex h-10 w-10 transform flex-col items-center justify-center overflow-hidden transition-all duration-500"
          >
            <div
              class="-translate-y-5 transform overflow-hidden transition-all duration-150 translate-y-3"
            >
              <svg
                class="h-6 w-6 animate-bounce text-white"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 15l7-7 7 7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <div
              class="flex h-[20px] w-[20px] origin-center -translate-y-3 transform flex-col justify-between overflow-hidden transition-all duration-300"
            >
              <div
                class="mb-1.5 h-[2px] w-7 origin-left transform bg-white transition-all duration-300 translate-y-6"
              />
              <div
                class="mb-1.5 h-[2px] w-7 transform rounded bg-white transition-all delay-75 duration-300 translate-y-6"
              />
              <div
                class="h-[2px] w-7 origin-left transform bg-white transition-all delay-100 duration-300 translate-y-6"
              />
            </div>
          </div>
        </button>
        <nav
          class="absolute top-14 right-0 h-0 w-full overflow-hidden bg-slate-700 text-white shadow-xl transition-all h-[128px]"
        >
          <ul>
            <li>
              <a
                class="block px-4 py-5"
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <a
                class="block px-4 py-5"
                href="/resume"
              >
                Resume
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <nav
        class="hidden md:block"
      >
        <ul
          class="flex gap-6"
        >
          <li
            class="flex flex-col items-center"
          >
            <a
              class="group block p-1 text-xl text-white transition duration-75"
              href="/blog"
            >
              <span>
                Blog
              </span>
              <span
                class="block h-0.5 max-w-0 bg-white transition-all duration-500 group-hover:max-w-full"
              />
            </a>
          </li>
          <li
            class="flex flex-col items-center"
          >
            <a
              class="group block p-1 text-xl text-white transition duration-75"
              href="/resume"
            >
              <span>
                Resume
              </span>
              <span
                class="block h-0.5 max-w-0 bg-white transition-all duration-500 group-hover:max-w-full"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
